### PR TITLE
Support for client IDs

### PR DESCRIPTION
**Pull request by Ballasi**
Added optional field `client_id` and `broker_client_id` in configuration files for MQTT authentification.
Also did a few identation correction and added client identification in the README.md file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,15 +35,17 @@ And another yaml file with basic info:
 
 ```yaml
 broker_host: tcp://broker.host:1883
+broker_client_id: client_id
```
Comment (phoenix-mstu) [inline README.md:38]: Why do you need a fixed clientId for broker connection?
Usually it should be a random string:
https://stackoverflow.com/questions/27174271/what-is-the-clientid-needed-for

File: README.md [hunk 2]
```diff
@@ -35,15 +35,17 @@ And another yaml file with basic info:
...
 broker_login: login
 broker_password: pass
 
 clients:
 
-  - login: some_login
+  - client_id: some_id
```
Comment (phoenix-mstu) [inline README.md:44]: Can we use client_id_prefix ? As all your clients probably change their ClientId in every connection (most of clients do)

File: internal/client/client_connection.go [hunk 3]
```diff
@@ -74,9 +74,9 @@ func (c *client) sendToBroker(MessageID uint16, topic string, qos byte, retained
 	}
 }
 
-func (c *client) login(username, password string) bool {
+func (c *client) login(client_id, username, password string) bool {
 	for _, conf := range c.configs {
-		if conf.Login == username && conf.Password == password {
+		if conf.ClientID == client_id && conf.Login == username && conf.Password == password {
```
Comment (phoenix-mstu) [inline internal/client/client_connection.go:79]: ClientId should not be used in authentication. It doesn't correspond MQTT specs https://www.hivemq.com/blog/mqtt-essentials-part-3-client-broker-connection-establishment/
Why do you need it?

**Review by phoenix-mstu — COMMENTED** [review]
Let's work on specs of the new config format first.
I suggest you not to write code until we're both agree with the new specs.

A good start would be to create a config_format.md file and describe all the config features which are currently supported.

**Comment by Ballasi** [comment]
Sorry! I was unaware client IDs was supposed to be used this way.

My MQTT server refers to client IDs as "credentials" alongside username and password so I assumed it was for authentication and my MQTT client was using predefined client IDs that did not change for each new connections. I will try to investigate this on my side why it works this way as it seems odd according to the specs. Thanks for the review, I am interested in talking about the specs.
You are free to close the PR for now!

FYI, I don't have a Telegram account.

**Pull request closed without merging**